### PR TITLE
Implement search for agency dashboard - mobile only

### PR DESCRIPTION
**Pull request by yashwin**
#### Changes proposed in this Pull Request

* This PR adds search to the agency dashboard only in the small screen view(<960px) along with fetching animation of cards while searching.
* This also adds a condition that makes sure the root component(`<SiteOverview />`) is not rendered if the agency dashboard feature flag is not enabled. This will guarantee that no API calls are made when a user hits `/dashboard` manually and the feature is not enabled.

#### Testing instructions

**Prerequisites**

Since this change is made specifically for agencies, you will have to set yourself(partner) as an agency - 2c49b-pb

**Instructions**

1. Run `git checkout add/implement-search-for-agency-dashboard` and `yarn start-jetpack-cloud`
2. Visit http://jetpack.cloud.localhost:3000/, and you'll be redirected to http://jetpack.cloud.localhost:3000/dashboard
4. Switch to small screen view(<960px) using the dev tool(inspect).
5. Verify that search is visible only on small screens(<960px) and hidden on large screens(>960px) 
6. Try entering some text in the search box and verify the results are appearing as expected.
7. Verify that the URL is appended with `?s={searchterm)` and the API call is being made(Since we have a delay(debounce with 300ms) in search, there won't be an API call on every key entered unless there is a delay between them) with params `?q=${searchterm}`

**Screenshots**

Screenshot 2022-05-19 at 6 39 50 PM

Screenshot 2022-05-19 at 6 40 14 PM

Screenshot 2022-05-19 at 6 40 22 PM

Large screen view(no search shown)

Screenshot 2022-05-20 at 11 32 33 AM

Related to 1202076982646589-as-1202122273254236

### CODE REVIEW

**Comment by robertf4** [comment]
It looks like the full list of sites is loaded with the page and searching then makes a new query to the server to get another truncated list. For this case doing the search as a simple filter on the client seems like it would give a better user experience since the results would be nearly instantaneous. 

If a user has enough sites to paginate the response then we would have to do the search as currently implemented, but it doesn't seem like the endpoint currently paginates and I suspect a very small fraction of users would benefit from pagination.

What are your thoughts?

**Comment by yashwin** [comment]
> It looks like the full list of sites is loaded with the page and searching then makes a new query to the server to get another truncated list. For this case doing the search as a simple filter on the client seems like it would give a better user experience since the results would be nearly instantaneous.
> 
> If a user has enough sites to paginate the response then we would have to do the search as currently implemented, but it doesn't seem like the endpoint currently paginates and I suspect a very small fraction of users would benefit from pagination.
> 
> What are your thoughts?

Hello @robertf4. Thanks for the review. Since this is a mock API that returns just a few items, you don't see pagination or search implemented. But, the actual API will have pagination implemented.

**Comment by yashwin** [comment]
Since we now have the API to get sites for agencies ready, we have integrated it and we are no longer using the mock API, but still, the results will not be according to the search term since the backend support for search is not yet implemented.

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-content/index.tsx [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
 import SiteTable from '../site-table';
 import { formatSites } from '../utils';
 import type { ReactElement } from 'react';
 
 import './style.scss';
 
-export default function SiteContent(): ReactElement {
-	const translate = useTranslate();
+interface Props {
+	data: Array< any > | undefined;
```
Comment (vitozev) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-content/index.tsx:12]: I'd suggest creating a new task and introducing a TypeScript interface for the site object.

This way, we can provide a much better type hinting for the available properties (e.g. `site.awaiting_plugin_updates`), and here, we can also replace `Array< any >` with `Array< AgencySite >` (or something similar), which makes much more sense.

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-search/index.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { useTranslate } from 'i18n-calypso';
+import { ReactElement } from 'react';
```
Comment (vitozev) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-search/index.tsx]: Minor: This could be `import type`

**Review by vitozev — APPROVED** [review]
LGTM!

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-content/index.tsx (hunk 1 above)
Comment (yashwin) [reply client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-content/index.tsx:12]: Done ✅ 

1202076982646589-as-1202346095941644

**Comment by yashwin** [comment]
@vitozev Thanks for the review, I have addressed your comments.

**Pull request merged**